From bf96e1b2f48eab26c4a0c2a0903d9d7b9a311a1d Mon Sep 17 00:00:00 2001
From: Michael Heimpold <michael.heimpold@i2se.com>
Date: Tue, 18 Dec 2018 14:47:16 +0100
Subject: [PATCH 1/9] Check calloc's return value before using the pointer

If calloc fails, bail out immediately instead of trying to
use the NULL pointer.

Signed-off-by: Michael Heimpold <michael.heimpold@i2se.com>
Cc: Michael Heimpold <mhei@heimpold.de>
---
 lsmmc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lsmmc.c b/lsmmc.c
index 9737b37..e514c83 100644
--- a/lsmmc.c
+++ b/lsmmc.c
@@ -374,6 +374,8 @@ char *to_binstr(char *hexstr)
 	char *binstr;
 
 	binstr = calloc(strlen(hexstr) * 4 + 1, sizeof(char));
+	if (!binstr)
+		return NULL;
 
 	while (hexstr && *hexstr != '\0') {
 		if (!isxdigit(*hexstr))
-- 
2.17.1

